3MI0800270/2023-03-07/zad5.cpp

```diff
@@ -0,0 +1,335 @@
+#include <iostream>
+#include <fstream>
+#include <cstring>
+using namespace std;
+
+const char FILE_PATH[10] = "zad5.txt";
+
+struct User {
+    char username[33];
+    char password[33];
+    char email[33];
+};
+
+int getTableSize(const char* filepath) {
+    ifstream file(filepath);
+    int count = 0;
+    while (!file.eof()) {
+        char ch = file.get();
+        if (ch == '\n') {
+            count++;
+        }
+    }
+    file.close();
+    return count;
+}
+
+void readUser(User& user, ifstream& file) {
+    char str[100];
+    char* ptr = str;
+
+    if (!file.eof()) {
+        file.getline(str, 99, '\n');
+    }
+    char username[33];
+    char* ptrT = username;
+    while (*ptr != ',') {
+        *ptrT = *ptr;
+        ptrT++;
+        ptr++;
+    }
+    *ptrT = '\0';
+    ptr++;
+    char password[33];
+    ptrT = password;
+    while (*ptr != ',') {
+        *ptrT = *ptr;
+        ptrT++;
+        ptr++;
+    }
+    *ptrT = '\0';
+    ptr++;
+    char email[33];
+    ptrT = email;
+    while (*ptr != '\0') {
+        *ptrT = *ptr;
+        ptrT++;
+        ptr++;
+    }
+    *ptrT = '\0';
+
+    strcpy(user.username, username);
+    strcpy(user.password, password);
+    strcpy(user.email, email);
+}
+
+bool isUsernameFree(const char* username, const char* filepath) {
+    int size = getTableSize(filepath);
```

Comment:
Btw - it's a bit a-logical what you're doing here - you read the file completely in order to determine the number of lines in it. Then you read it once again in order to do something with the data. What if this file was 35 GB in size ?